mroderick/plete
# ARIA activedescendant, label

#### Purpose (TL;DR)

To enable screen reader software and assistive ARIA aware technologies to announce each item as highlighted prior to selection.
Fixes #77 

#### Background (Problem in detail) 

Detailed in #77 

#### Solution

Uses additional ARIA attributes to announce choices prior to selection, differentiating from selection announcement.
Supporting DOM changes. (likely "choice" constant will need to be configurable).

#### Other solutions considered

Please see ticket.

* ARIA live region (did not announce on selection).

#### How to verify - mandatory

1. Check out this branch
2. `npm ci`
3. `npm run start`
4. use http://localhost:8080/
5. download browser extension for Google Chrome (or any screenreader) [link](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn)

#### Contributor declaration

- [x] I've read and followed the [contributing document](CONTRIBUTING.md).
- [x] The code I am submitting is mine to submit and is done so under the license of this repository.
- [x] Where features have been changed, there are tests and documentation to accompany the entire contribution.

* All tests remained passing as the behavior of these attributes is not owned by plete.
* Documentation should not be necessary as this is a transparent change.
* Please let me know. I expect the first-cut is an improvement, but may need further improvement.
